src/status_im/chat/models/message.cljs

```diff
@@ -198,14 +199,13 @@
             (chat-loading/group-chat-messages chat-id (get chat->message chat-id))))
 
 (defn- add-to-chat?
-  [{:keys [db]} {:keys [chat-id clock-value message-id from] :as message}]
-  (let [deduplication-id (messages-store/deduplication-id from chat-id clock-value)
-        {:keys [deleted-at-clock-value messages not-loaded-message-ids deduplication-ids]}
+  [{:keys [db]} {:keys [chat-id clock-value message-id from]}]
+  (let [{:keys [deleted-at-clock-value messages not-loaded-message-ids]}
         (get-in db [:chats chat-id])]
     (not (or (get messages message-id)
              (get not-loaded-message-ids message-id)
-             (get deduplication-ids deduplication-id)
-             (>= deleted-at-clock-value clock-value)))))
+             (>= deleted-at-clock-value clock-value)
+             (messages-store/message-exists? chat-id)))))
```

Comment:
shouldn't it be message-id? also does it mean we trade initial loading time for a query for each message received ?

Reply:
oh right

Reply:
fixed.
Yes, we do a query for each message and that's totally fine comparing to iterating all messages in realm db. Also `objectForPrimaryKey` is quite fast.